backend/lib/services/members/MemberManager.js

```diff
@@ -187,35 +187,38 @@ class MemberManager {
     this.subjectList.delete(item.id)
   }
 
-  async deleteServiceAccountSecret (item) {
+  async deleteServiceAccountSecrets (item) {
     const { namespace } = Member.parseUsername(item.id)
     if (namespace !== this.namespace) {
       throw new UnprocessableEntity('It is not possible to modify a ServiceAccount from another namespace')
     }
 
-    const name = _
+    const names = _
       .chain(item)
       .get('extensions.secrets', [])
-      .tap(secrets => {
-        if (secrets.length > 1) {
-          throw new UnprocessableEntity(`ServiceAccount ${namespace} has more than one secret`)
-        }
-      })
-      .head()
-      .get('name')
+      .map('name')
       .value()
-    await this.client.core.secrets.delete(namespace, name)
+
+    _.forEach(names, async name => {
```

Comment:
```js
const results = await _
      .chain(item)
      .get('extensions.secrets')
      .map(name => this.client.core.secrets.delete(namespace, name))
      .thru(Promise.allSettled)
      .value()
const errors = _
      .chain(results)
      .filter(['status', 'rejected'])
      .map('reason')
      .filter(err => !isHttpError(err) || err.statusCode !== 404)
      .value()
if (errors.length) {
     throw errors[0] // or whatever we want to do in this case with the errors
}
```

Reply:
If an error occurs inside the `forEach` loop it will cause an `unhandledRejection` error which leads to process exit. All the function will be called at the same time. The usage of async / await in this case has no effect at all. It would have been much better to use 
```js
for (const name of names) {
  await this.client.core.secrets.delete(namespace, name)
}
```
instead to run the delete one by one and forward the rejection to the outer function.